### PR TITLE
tickets/DM-40289: Add support for extra annotations on pod

### PR DESCRIPTION
**Pull request by athornton**
Needed for T&S and their bridged-networking magic.

### CODE REVIEW

**Comment by athornton** [comment]
There's some commentary about why we need to ignore typing for two tests.  Strangely, I can't reproduce the bug with a minimal class that does the same thing, so it's at least a little subtle.

**Comment by athornton** [comment]
Here's the test case which fails to fail like jupyterlab-controller does.

First, the requirements.txt for the virtualenv you should run it in:
```
mypy==1.4
pytest
```

And here's `bug_test.py`, which should be run with `pytest`:
```
import pytest
from enum import Enum


class MyEnum(Enum):
    INITIAL = "initial"
    FINAL = "final"


class EnumFlipper:
    def __init__(self) -> None:
        self.state = MyEnum.INITIAL

    def finalize(self) -> None:
        self.state = MyEnum.FINAL


def test_bug() -> None:
    obj = EnumFlipper()
    assert obj.state == MyEnum.INITIAL
    obj.finalize()
    assert obj.state == MyEnum.FINAL
```

But this behaves exactly as you'd expect:
```
adam@m1-wired:~/Documents/src/mypy-bug$ pytest .
============================= test session starts ==============================
platform darwin -- Python 3.11.4, pytest-7.4.0, pluggy-1.2.0
rootdir: /Users/adam/Documents/src/mypy-bug
collected 1 item

bug_test.py .                                                            [100%]

============================== 1 passed in 0.00s ===============================
(mypybug)
adam@m1-wired:~/Documents/src/mypy-bug$ mypy .
Success: no issues found in 1 source file
(mypybug)
```

**Comment by rra** [comment]
Oh, even more likely than it being all in one file, your example isn't a dataclass, which I suspect changes the way mypy thinks about attributes. (Classes like dataclasses and Pydantic models are more likely to be immutable, and thus calls to apparently unrelated methods are less likely to change the attribute values than a regular class.)

**Comment by rra** [comment]
Oh, and sorry, that was ambiguous: by drop changes to the test suite, I just meant the ones working around the mypy limitation, not the new tests you added (which are fine).

File: requirements/dev.in [hunk 1]
```diff
@@ -12,6 +12,7 @@ asgi-lifespan
 coverage[toml]
 httpx-sse
 mypy
+# mypy==1.3
```
Comment (rra) [inline requirements/dev.in:15]: This can now be removed.

File: tests/models/domain/rspimage_test.py [hunk 2]
```diff
@@ -155,6 +155,7 @@ def test_collection() -> None:
     SystemRandom().shuffle(shuffled_images)
     assert images[0].aliases == set()
     collection = RSPImageCollection(shuffled_images)
+    # Same mypy error as above in test_resolve_alias
```
Comment (rra) [inline tests/models/domain/rspimage_test.py:158]: I think this comment is no longer relevant.

**Review by rra — APPROVED** [review]

**Pull request merged**